Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/BreadCrumbs.cpp

```diff
@@ -82,9 +81,38 @@ namespace AzQtComponents
         // clean up the path to use all the first separator in the list of separators
         m_currentPath.replace(g_windowsSeparator, g_separator);
 
+        // update internals
+        m_currentPathSize = m_currentPath.split(g_separator, Qt::SkipEmptyParts).size();
+        m_currentPathIcons.resize(m_currentPathSize);
+
         fillLabel();
     }
 
+    void BreadCrumbs::setDefaultIcon(const QString& icon)
+    {
+        m_defaultIcon = icon;
+    }
+
+    void BreadCrumbs::setIconAt(int index, const QString& icon)
+    {
+        if (index < 0 || index >= m_currentPathSize)
+        {
+            return;
+        }
+
+        m_currentPathIcons[index] = icon;
+    }
+    
+    QIcon BreadCrumbs::iconAt(int index)
+    {
+        if (index < 0 || index >= m_currentPathSize)
+        {
+            return QIcon(m_defaultIcon);
+        }
+
+        return !m_currentPathIcons[index].isNull() ? QIcon(m_currentPathIcons[index]) : QIcon(m_defaultIcon);
```

Comment:
Maybe add the null check here as well so you don't need the ternary at the bottom
```suggestion
        if (index < 0 || index >= m_currentPathSize || m_currentPathIcons[index].isNull())
        {
            return QIcon(m_defaultIcon);
        }
        else
        {
            return QIcon(m_currentPathIcons[index];
        }
```

Reply:
Yeah that's much more readable, thanks :D